crates/orchestrator/src/lib.rs

```diff
@@ -341,14 +363,52 @@ where
     // TODO ED Add a map to verify which nodes have posted they're ready
     fn post_ready(&mut self) -> Result<(), ServerError> {
         self.nodes_connected += 1;
+
         println!("Nodes connected: {}", self.nodes_connected);
+
         // i.e. nodes_connected >= num_nodes_with_stake * (start_threshold.0 / start_threshold.1)
         if self.nodes_connected * self.config.config.start_threshold.1
             >= (self.config.config.num_nodes_with_stake.get() as u64)
                 * self.config.config.start_threshold.0
         {
+            self.accepting_connections = false;
+            self.manual_start_allowed = false;
+            self.start = true;
+        }
+
+        Ok(())
+    }
+
+    /// Manually start the network
+    fn post_manual_start(&mut self, password_bytes: Vec<u8>) -> Result<(), ServerError> {
+        if !self.manual_start_allowed {
+            return Err(ServerError {
+            status: tide_disco::StatusCode::Forbidden,
+            message: "Configs have already been distributed to nodes, and the network can no longer be started manually.".to_string(),
+          });
+        }
+
+        let password = String::from_utf8(password_bytes)
+            .expect("Failed to decode raw password as UTF-8 string.");
+
+        if self.pub_posted.len() > 1 {
+            self.config.config.num_nodes_with_stake =
+                std::num::NonZeroUsize::new(self.pub_posted.len())
+                    .expect("Failed to convert to NonZeroUsize; this should be impossible.");
+        } else {
+            return Err(ServerError {
+                status: tide_disco::StatusCode::Forbidden,
+                message: format!("We cannot manually start the network, because we have {} public keys registered.", self.pub_posted.len())
+            });
+        }
+
+        if self.config.manual_start_password == Some(password) {
+            self.accepting_connections = false;
+            self.manual_start_allowed = false;
+            self.peer_pub_ready = true;
             self.start = true;
```

Comment:
I don't think we want this here.  In my view, a manual restart should stop the orchestrator from registering keys and make the orchestrator start sending out the config.  We still want all the nodes that have registered keys to post that they are "ready" to start before setting self.start to true.

Reply:
Removed!

Also just reorganized `post_manual_start` a bit. We now return immediately if the password is wrong or it's too late to manually start. If everything is fine, we set `num_nodes_with_stake` to `self.pub_posted.len()` and signal `peer_pub_ready`.